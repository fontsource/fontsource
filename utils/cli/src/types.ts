export interface CLIOptions {
	test?: boolean;
	out?: string;
}

export interface BuildOptions {
	dir: string;
	tmpDir: string;
	force: boolean;
	isVariable: boolean;
}

interface Axes {
	[axesType: string]: {
		default: string;
		min: string;
		max: string;
		step: string;
	};
}

export type VariableMetadata = boolean | Axes;

export type CategoryNames =
	| 'sans-serif'
	| 'serif'
	| 'display'
	| 'handwriting'
	| 'monospace'
	| 'icons'
	| 'other';

export type TypeNames = 'google' | 'league' | 'icons' | 'other';

export interface Metadata {
	id: string;
	family: string;
	subsets: string[];
	weights: number[];
	styles: string[];
	defSubset: string;
	variable: VariableMetadata;
	lastModified: string;
	version: string;
	category: CategoryNames;
	license: {
		type: string;
		url: string;
		attribution: string;
	};
	source: string;
	type: TypeNames;
}
