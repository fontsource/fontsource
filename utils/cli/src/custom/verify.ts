/* eslint-disable consistent-return */
import { cancel, group, intro, outro,text } from '@clack/prompts';
import fs from 'fs-extra';
import path from 'pathe';
import colors from 'picocolors';

import { Metadata } from '../types';

export const verifyFilenames = async (metadata: Metadata, dir: string) => {
	// Read all the filenames in the files directory
	const filenames = await fs.readdir(path.join(dir, 'files'));

	// Make sure filenames is not empty
	if (filenames.length === 0) {
		throw new Error('No files found in the files directory.');
	}

	// Generate a list of expected filenames
	const expectedFilenames: string[] = [];

	// Iterate over all subsets
	for (const subset of metadata.subsets) {
		// Iterate over all weights
		for (const weight of metadata.weights) {
			// Iterate over all styles
			for (const style of metadata.styles) {
				// Add the expected filename to the list
				expectedFilenames.push(`${metadata.id}-${subset}-${weight}-${style}.woff2`, `${metadata.id}-${subset}-${weight}-${style}.woff`);
			}
		}
	}


	// Check if all expected filenames are present and show all missing or non-matching filenames
	const missingFilenames = expectedFilenames.filter((filename) => !filenames.includes(filename));
	if (missingFilenames.length > 0) {
		throw new Error(`The following files are missing: ${missingFilenames.join(', ')}\n\n\tPlease ensure the file names match the format "${metadata.id}-subset-weight-style.extension"\n\tExample: "${metadata.id}-latin-400-normal.woff2" or "${metadata.id}-latin-ext-700-italic.woff"`);
	}

	// Check if all filenames are expected and show all non-matching filenames
	const nonMatchingFilenames = filenames.filter((filename) => !expectedFilenames.includes(filename));
	if (nonMatchingFilenames.length > 0) {
		throw new Error(`The following files are not expected: ${nonMatchingFilenames.join(', ')}\n\n\tPlease ensure the file names match the format "${metadata.id}-subset-weight-style.extension"\n\tExample: "${metadata.id}-latin-400-normal.woff2" or "${metadata.id}-latin-ext-700-italic.woff"`);
	}


};


export const verify = async (): Promise<void> => {
	intro(colors.cyan(colors.bold('fontsource')));
	const cfg = await group({
		id: () => text({
			message: colors.bold('What is the ID of the font?'),
			placeholder: 'noto-sans-jp',
			validate(value) {
				if (!value) return 'Please enter an ID.';
			}
		}),
	});

	// Check if the directory does not exist
	if (!(await fs.pathExists(`./${cfg.id}/files`))) {
		cancel('The directory does not exist.');
		return;

	}

	// Check if the metadata.json file exists
	if (!(await fs.pathExists(`./${cfg.id}/metadata.json`))) {
		cancel('The metadata.json file does not exist.');
		return;
	}

	// Check if LICENSE file exists
	if (!(await fs.pathExists(`./${cfg.id}/LICENSE`))) {
		cancel('The LICENSE file does not exist.');
		return;
	}

	// Verify filenames
	try {
		const dir = `./${cfg.id}`;
		// Read metadata.json
		const metadata: Metadata = await fs.readJson(path.join(dir, 'metadata.json'));
		await verifyFilenames(metadata, dir);
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	} catch (error: any) {
		cancel(error.message);
		return;
	}

	outro(colors.green('All checks passed! Feel free to send a PR over to the main repo adding the package to the appropriate fonts directory.'));
};


