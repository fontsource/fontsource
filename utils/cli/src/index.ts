export {
	APIIconStatic,
	APIIconVariable,
	APILicense,
	APIRegistry,
	APIv1,
	APIv2,
	APIVariable,
} from 'google-font-metadata';
