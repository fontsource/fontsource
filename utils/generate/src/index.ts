import type { FontMetadata, FontObject } from './types';
import { findBoundaries } from './utils';

const generateSingle = (font: FontObject) => {
	const { family, style, display, weight, variable, src, unicodeRange, comment, spacer = '\n  ' } = font;
	// If variable, modify output
	const { wght, stretch, slnt } = variable ?? {};
	let result = '@font-face {';
	result += `${spacer}font-family: '${family}';`;

	// If slnt is present, switch to oblique style
	result += `${spacer}font-style: ${slnt ? `oblique ${slnt.max * -1}deg ${slnt.min * -1}deg` : style};`;
	result += `${spacer}font-display: ${display};`;
	// If variable wght is present, use min max wght vals
	result += `${spacer}font-weight: ${wght ? findBoundaries(wght).join(' ') : weight};`;

	if (stretch)
		result += `${spacer}font-stretch: ${stretch.min}% ${stretch.max}%;`;

	// Merges all formats into a single src
	result += `${spacer}src: ${src.map(({ url, format }) => `url(${url}) format('${format}')`).join(', ')};`;

	if (unicodeRange)
		result += `${spacer}unicode-range: ${unicodeRange};`;

	if (comment)
		result = `/* ${comment} */\n${result}`;

	return `${result}\n}`;
};

const generateMulti = (metadata: FontMetadata) => {
	const { family, styles, display, weights, variable, path = '', subsets, unicodeRange, formats } = metadata;
	const id = metadata.id ?? family.toLowerCase().replace(/\s/g, '-');
	let result = '';
	for (const style of styles) {
		for (const weight of weights) {
			for (const subset of subsets) {
				const fontObject: FontObject = {
					family,
					style,
					display,
					weight,
					variable,
					src: formats.map(format => ({ url: `${path}${id}-${subset}-${weight}-${style}.${format}`, format })),
					comment: subset
				};

				if (unicodeRange)
					fontObject.unicodeRange = unicodeRange[subset];

				result += `${generateSingle(fontObject)}\n\n`;
			}
		}
	}
	// Slice last 2 \n if no minify
	return result.slice(0, -2);
};

export { generateMulti, generateSingle };