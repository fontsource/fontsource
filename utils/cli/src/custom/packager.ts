/* eslint-disable no-await-in-loop */
import { generateFontFace } from '@fontsource-utils/generate';
import fs from 'fs-extra';
import path from 'pathe';

import { Metadata } from '../types';
import { findClosest, makeFontFilePath } from '../utils';

export const packagerCustom = async (metadata: Metadata) => {
	const { id, family, subsets, weights, styles } = metadata;
	const dir = `./${id}`;

	// Find the weight for index.css in the case weight 400 does not exist.
  const indexWeight = findClosest(weights, 400);

	// Write the CSS files
	for (const subset of subsets) {
		// Arrays of CSS blocks to be concatenated
		const cssSubset: string[] = [];
		const cssSubsetItalic: string[] = [];

		for (const weight of weights) {
			for (const style of styles) {
				const fontObj = {
					family,
					style,
					display: 'swap',
					weight,
					src: [
						{
							url: makeFontFilePath(id, subset, weight, style, 'woff2'),
							format: 'woff2',
						},
						{
							url: makeFontFilePath(id, subset, weight, style, 'woff'),
							format: 'woff',
						},
					],
					comment: `${id}-${subset}-${weight}-${style} CUSTOM`,
					displayVar: true,
				};
				// This takes in a font object and returns an @font-face block
				const css = generateFontFace(fontObj);

				// Needed to differentiate filenames
				if (style === 'normal') {
					await fs.writeFile(path.join(dir,`${subset}-${weight}.css`), css);
					cssSubset.push(css);
				} else {
					await fs.writeFile(path.join(dir,`${subset}-${weight}-${style}.css`), css);
					cssSubsetItalic.push(css);
				}
			}

			if (weight === indexWeight) {
				await fs.writeFile(path.join(dir,'index.css'), cssSubset.join('\n\n'));
			}

			await fs.writeFile(path.join(dir,`${subset}.css`), cssSubset.join('\n\n'));

			// If there are italic styles for a subset
			if (cssSubsetItalic.length > 0) {
				await fs.writeFile(path.join(dir,`${subset}-italic.css`), cssSubsetItalic.join('\n\n'));
			}
		}
	}
};
