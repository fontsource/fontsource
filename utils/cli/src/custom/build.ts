import fs from 'fs-extra';
import stringify from 'json-stringify-pretty-compact';
import path from 'pathe';

import { sassMetadata } from '../sass/metadata';
import { sassMixins } from '../sass/mixins';
import { changelog } from '../templates/changelog';
import { packageJson } from '../templates/package';
import { readme } from '../templates/readme';
import type { Metadata } from '../types';
import { packagerCustom } from './packager';

export const buildCustom = async (metadata: Metadata) => {
	// Create the directory for the font
	const dir = `./${metadata.id}`;
	fs.ensureDirSync(dir);
	fs.ensureDirSync(path.join(dir, '/files'));

	// Generate CSS files
	await packagerCustom(metadata);

	// Write metadata.scss
	await fs.mkdir(path.join(dir, 'scss'));
	await fs.writeFile(
		path.join(dir, 'scss/metadata.scss'),
		sassMetadata(metadata, {})
	);

	// Write mixins.scss
	await fs.writeFile(path.join(dir, 'scss/mixins.scss'), sassMixins);

	// Write README.md
	await fs.writeFile(path.join(dir, 'README.md'), readme(metadata));

	// Write metadata.json
	await fs.writeFile(path.join(dir, 'metadata.json'), stringify(metadata));

	// Write unicode.json
	await fs.writeFile(path.join(dir, 'unicode.json'), stringify({}));

	// Write CHANGELOG.md
	await fs.writeFile(path.join(dir, 'CHANGELOG.md'), changelog);

	// Write package.json
	await packageJson(metadata, dir);
};
