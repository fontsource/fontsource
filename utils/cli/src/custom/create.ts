/* eslint-disable consistent-return */
import { cancel, confirm, group, intro, multiselect, outro, select, text } from '@clack/prompts';
import consola from 'consola';
import colors from 'picocolors';

import type { Metadata } from '../types';
import { licenseShort } from '../utils';
import { buildCustom } from './build';

export const create = async () => {
	intro(colors.cyan(colors.bold('fontsource')));
	const cfg = await group({
		name: () => text({
			message: colors.bold('What is the name of the font?'),
			placeholder: 'Noto Sans JP',
			validate(value) {
				if (!value) return 'Please enter a name';
			}
		}),
		subsets: () => text({
			message: colors.bold('What are the subsets of the font? (separate by commas)'),
			placeholder: 'latin, latin-ext, japanese',
			validate(value) {
				if (!value) return 'Please enter at least one subset';
			}
		}),
		weights: () => text({
			message: colors.bold('What are the weights of the font? (separate by commas)'),
			placeholder: '100, 200, 300, 400, 500, 600, 700, 800, 900',
			validate(value) {
				if (!value) return 'Please enter at least one weight';
				// Split array and check if all values are numbers
				const weights = value.split(',').map(Number);
				if (weights.some((weight) => Number.isNaN(weight))) return 'Please enter only numbers';
			}
		}),
		styles: () => multiselect({
			message: colors.bold('What are the styles of the font?'),
			options: [
				{ value: 'normal', label: 'Normal' },
				{ value: 'italic', label: 'Italic' }
			],
			required: true
		}),
		version: () => text({
			message: colors.bold('What is the version of the font?'),
			placeholder: 'v1.0',
			validate(value) {
				if (!value) return 'Please enter a version';
			}
		}),
		category: () => select({
			message: colors.bold('What is the category of the font?'),
			options: [
				{ value: 'sans-serif', label: 'Sans Serif' },
				{ value: 'serif', label: 'Serif' },
				{ value: 'display', label: 'Display' },
				{ value: 'monospace', label: 'Monospace' },
				{ value: 'handwriting', label: 'Handwriting' },
				{ value: 'icons', label: 'Icons' },
				{ value: 'other', label: 'Other' }
			]
		}),
		license: () => select({
			message: colors.bold('What is the license of the font?'),
			options: [
				{ value: 'sil open font license, 1.1', label: 'SIL Open Font License (OFL-1.1)' },
				{ value: 'apache license, version 2.0', label: 'Apache License 2.0 (Apache-2.0)' },
				{ value: 'ubuntu font license, 1.0', label: 'Ubuntu Font License (UFL-1.0)' },
				{ value: 'mit', label: 'MIT License' },
				{ value: 'cc0-1.0', label: 'CC0-1.0 License' },
				{ value: 'other', label: 'Other License', hint: 'Please make an issue verifying if this is usable! We are likely to reject PRs that do not match the above licenses.' }
			]
		}),
		licenseUrl: () => text({
			message: colors.bold('What is the URL of the license?'),
			validate(value) {
				if (!value) return 'Please enter a URL';
			}
		}),
		licenseAttribution: () => text({
			message: colors.bold('What is the author name or attribution of the font?'),
			validate(value) {
				if (!value) return 'Please enter a name or attribution';
			}
		}),
		sourceUrl: () => text({
			message: colors.bold('What is the URL of the source files? GitHub repositories are preferred'),
			validate(value) {
				if (!value) return 'Please enter a URL';
			}
		}),
	},
	{
    onCancel: () => {
      cancel('Package creation cancelled');
      // eslint-disable-next-line unicorn/no-process-exit
      process.exit(0);
    },
		});

	const metadata: Metadata = {
		id: cfg.name.toLowerCase().replace(/ /g, '-'),
		family: cfg.name,
		subsets: cfg.subsets.split(',').map((subset) => subset.trim()),
		weights: cfg.weights.split(',').map((weight) => Number.parseInt(weight.trim(), 10)),
		styles: cfg.styles,
		defSubset: cfg.subsets.split(',')[0].trim(),
		variable: false,
		lastModified: new Date().toISOString().split('T')[0], // YYYY-MM-DD
		version: cfg.version,
		category: cfg.category,
		license: {
			type: licenseShort(cfg.license) ?? cfg.license,
			url: cfg.licenseUrl,
			attribution: cfg.licenseAttribution
		},
		source: cfg.sourceUrl,
		type: 'other'
	};

	const confirmCreate = await confirm({
		message: colors.bold(`Create package in current directory? ./${metadata.id}`)
	});

	if (!confirmCreate) {
		cancel('Package creation cancelled.');
		// eslint-disable-next-line unicorn/no-process-exit
		process.exit(0);
	}

	try {
		await buildCustom(metadata);
	} catch (error) {
		cancel('Package creation failed.');
		consola.error(error);
		return;
	}

	outro(colors.green(`You're all set!\n\n\tNow copy all the font files into the files directory and run "fontsource create-verify" to verify your package.\n\n\tPlease ensure the file names match the format "${metadata.id}-subset-weight-style.extension"\n\tExample: "${metadata.id}-latin-400-normal.woff2" or "${metadata.id}-latin-ext-700-italic.woff"\n\n\tPlease also copy the appropriate LICENSE file from your license URL to the root directory of the package.`));
};
