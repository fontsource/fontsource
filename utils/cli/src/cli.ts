import 'dotenv/config';

import { cac } from 'cac';
import consola from 'consola';
import {
	fetchAPI,
	fetchVariable,
	generateAxis,
	parseLicenses,
	parsev1,
	parsev2,
	parseVariable } from 'google-font-metadata';
import colors from 'picocolors';

import { version } from '../package.json';
import { create } from './custom/create';
import { verify } from './custom/verify';
import { processGoogle } from './google/queue';

const cli = cac('fontsource');

cli
	.command('fetch [key]', 'Fetch parsing metadata for all fonts')
	.option('-f, --force', 'Force parse all metadata')
	.action(async (key: string, options) => {
		try {
			const finalKey = key ?? process.env.GOOGLE_API_KEY;
			if (!finalKey) {
				throw new Error('No API key provided.');
			}
			if (options.force) {
				consola.info(
					`Parsing all metadata... ${colors.bold(colors.red('[FORCE]'))}`
				);
			} else {
				consola.info('Parsing all metadata...');
			}
			await Promise.all([fetchAPI(finalKey), fetchVariable()]);
			await parsev1(options.force, false);
			await parsev2(options.force, false);
			await generateAxis();
			await parseVariable(false);
			await parseLicenses();
		} catch (error) {
			consola.error(error);
		}
	});

cli
	.command('build [...fonts]', 'Build font packages')
	.option('-v, --variable', 'Only build variable fonts')
	.option('-f, --force', 'Force rebuild all packages')
	.option('-t, --test', 'Build test fonts only')
	.action(async (fonts: string[], options) => {
		try {
			consola.info(`Building packages... ${options.force ? colors.bold(colors.red('[FORCE]')) : ''}`);
			await processGoogle(options, fonts);
		} catch (error) {
			consola.error(error);
		}
	});

cli.command('create').action(async () => {
	try {
		await create();
	} catch (error) {
		consola.error(error);
	}
});

cli.command('create-verify').action(async () => {
	try {
		await verify();
	} catch (error) {
		consola.error(error);
	}
});

cli.help();
cli.version(version);

cli.parse();
