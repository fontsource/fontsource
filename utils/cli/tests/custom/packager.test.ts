import fs from 'fs-extra';
import { describe, expect, it, vi } from 'vitest';

import { packagerCustom } from '../../src/custom/packager';
import { Metadata } from '../../src/types';

vi.mock('fs-extra');

describe('packager custom', () => {
	it('should generate custom CSS successfully', async () => {
		const metadata: Metadata = {
			id: 'custom-font',
			family: 'Custom Font',
			variable: false,
			weights: [400,500],
			styles: ['normal', 'italic'],
			defSubset: 'latin',
			subsets: ['latin', 'latin-ext'],
			lastModified: '2021-01-01',
			version: '1.0.0',
			category: 'display',
			license: {
				type: 'OFL',
				url: 'license-url',
				attribution: 'Custom Font Author',
			},
			source: 'source-url',
			type: 'other',
		};

		await packagerCustom(metadata);
		expect(vi.mocked(fs.writeFile)).toMatchSnapshot();
	});
});
