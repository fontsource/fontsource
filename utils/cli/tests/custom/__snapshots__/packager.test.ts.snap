// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`packager custom > should generate custom CSS successfully 1`] = `
[MockFunction writeFile] {
  "calls": [
    [
      "custom-font/latin-400.css",
      "/* custom-font-latin-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-400-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-400-italic.css",
      "/* custom-font-latin-400-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-400-italic.woff2) format('woff2'), url(./files/custom-font-latin-400-italic.woff) format('woff');
}",
    ],
    [
      "custom-font/index.css",
      "/* custom-font-latin-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-400-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin.css",
      "/* custom-font-latin-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-400-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-italic.css",
      "/* custom-font-latin-400-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-400-italic.woff2) format('woff2'), url(./files/custom-font-latin-400-italic.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-500.css",
      "/* custom-font-latin-500-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-500-normal.woff2) format('woff2'), url(./files/custom-font-latin-500-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-500-italic.css",
      "/* custom-font-latin-500-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-500-italic.woff2) format('woff2'), url(./files/custom-font-latin-500-italic.woff) format('woff');
}",
    ],
    [
      "custom-font/latin.css",
      "/* custom-font-latin-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-400-normal.woff) format('woff');
}

/* custom-font-latin-500-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-500-normal.woff2) format('woff2'), url(./files/custom-font-latin-500-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-italic.css",
      "/* custom-font-latin-400-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-400-italic.woff2) format('woff2'), url(./files/custom-font-latin-400-italic.woff) format('woff');
}

/* custom-font-latin-500-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-500-italic.woff2) format('woff2'), url(./files/custom-font-latin-500-italic.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext-400.css",
      "/* custom-font-latin-ext-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-ext-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-ext-400-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext-400-italic.css",
      "/* custom-font-latin-ext-400-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-ext-400-italic.woff2) format('woff2'), url(./files/custom-font-latin-ext-400-italic.woff) format('woff');
}",
    ],
    [
      "custom-font/index.css",
      "/* custom-font-latin-ext-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-ext-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-ext-400-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext.css",
      "/* custom-font-latin-ext-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-ext-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-ext-400-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext-italic.css",
      "/* custom-font-latin-ext-400-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-ext-400-italic.woff2) format('woff2'), url(./files/custom-font-latin-ext-400-italic.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext-500.css",
      "/* custom-font-latin-ext-500-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-ext-500-normal.woff2) format('woff2'), url(./files/custom-font-latin-ext-500-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext-500-italic.css",
      "/* custom-font-latin-ext-500-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-ext-500-italic.woff2) format('woff2'), url(./files/custom-font-latin-ext-500-italic.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext.css",
      "/* custom-font-latin-ext-400-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-ext-400-normal.woff2) format('woff2'), url(./files/custom-font-latin-ext-400-normal.woff) format('woff');
}

/* custom-font-latin-ext-500-normal CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: normal;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-ext-500-normal.woff2) format('woff2'), url(./files/custom-font-latin-ext-500-normal.woff) format('woff');
}",
    ],
    [
      "custom-font/latin-ext-italic.css",
      "/* custom-font-latin-ext-400-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 400;
  src: url(./files/custom-font-latin-ext-400-italic.woff2) format('woff2'), url(./files/custom-font-latin-ext-400-italic.woff) format('woff');
}

/* custom-font-latin-ext-500-italic CUSTOM */
@font-face {
  font-family: 'Custom Font';
  font-style: italic;
  font-display: swap;
  font-weight: 500;
  src: url(./files/custom-font-latin-ext-500-italic.woff2) format('woff2'), url(./files/custom-font-latin-ext-500-italic.woff) format('woff');
}",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
